nucypher/blockchain/eth/agents.py

```diff
@@ -252,7 +257,10 @@ class StakingEscrowAgent(EthereumContractAgent):
         'finishUpgrade'
     )
 
-    DEFAULT_PAGINATION_SIZE: int = 30    # TODO: Use dynamic pagination size (see #1424)
+    DEFAULT_STAKER_PAGINATION_SIZE_LIGHT_NODE: int = int(os.environ.get(
```

Comment:
This pattern worries me. First, the pagination size feels like a more high-level setting; whoever calls `StakingEscrowAgent` should worry about setting it. Second, the low-level class's behavior is influenced by an envvar, which may lead to some hard to diagnose problems. The only other place where we do that is `ContractEventsThrottler`.

Reply:
The caller of `StakingEscrowAgent` can still proactively specify a `pagination_size` see `get_all_active_stakers()` which is the only place a pagination size is needed/used. If the user doesn't specify a pagination_size (i.e. `None`), only then is the default used.

We were already using a default value for light nodes in prior code because they act differently than regular nodes. However, the number was fixed and could not be modified meaning that if we were wrong on the value it couldn't be modified in the field without changing the code. To me, this was the best compromise between allowing the user to specify a pagination size, if not use a default, if the default failed then it could be modified proactively.

The goal here is not to have the env var ever set, but rather have it act as a fallback just in case, for some reason, our beliefs about what the pagination size should be are incorrect.

Regarding "hard to diagnose" problems, for extra help, we log when the default pagination size is used and its value. see L324.

Reply:
I don't mind the default, or that there's an envvar for the default, it just seems that this specific spot is too deep in the call stack to depend on external environment. I admit I can't think of a good solution at the moment.

> Regarding "hard to diagnose" problems, for extra help, we log when the default pagination size is used and its value.

It helps, but it's still a surprising side effect.